src/lib.rs

```diff
@@ -97,3 +99,9 @@ pub trait GenFn4<T>: Fn(&PermutationTable, &Point4<T>) -> T {}
 impl<T, F> GenFn2<T> for F where F: Fn(&PermutationTable, &Point2<T>) -> T, {}
 impl<T, F> GenFn3<T> for F where F: Fn(&PermutationTable, &Point3<T>) -> T, {}
 impl<T, F> GenFn4<T> for F where F: Fn(&PermutationTable, &Point4<T>) -> T, {}
+
+pub trait Module<T> {
```

Comment:
Still not sure about changing the name of Module to Source/ValueSource. Not all of these will 'source' a value. Some will modify their passed values, others won't modify the value, but modify the point instead.

A quick google search indicates that most of the other language implementations of procedural noise are using either Module, IModule, or ModuleBase as the common parent name.